demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/EventMapBehaviour.java

```diff
@@ -0,0 +1,470 @@
+
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(View.VISIBLE);
+
+        if (bottomSheetBehaviour != null) {
+            bottomSheetBehaviour.setDisplayedText(marker.getTitle());
+            if (bottomSheetBehaviour.isShown())
+                bottomSheetBehaviour.hide();
+            else
+                bottomSheetBehaviour.show();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * If set, hides the Bottom Sheet
+     * If set, hides the goToNavigatorView
+     * Makes forget the previously set currentFocusMarker
+     *
+     * @param latLng The position on the map where the user clicked
+     */
+    @Override
+    public void onMapClick(LatLng latLng) {
+        if (bottomSheetBehaviour != null)
+            bottomSheetBehaviour.hide();
+
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(INVISIBLE);
+
+        currentFocusMarker = null;
+    }
+
+    /**
+     * Called by a fragment of the application that returns the parameters to Create a new event and update the map
+     *
+     * @param pos         The position of the event to create
+     * @param description The description of the event to create
+     */
+    @Override
+    public void onEventReturn(LatLng pos, String description) {
+        GPSPosition eventPos = new GPSPosition(pos.latitude, pos.longitude);
+        Event event = new GenericEvent<>(eventPos, description);
+        if (onEventCreatedListeners != null && !onEventCreatedListeners.isEmpty())
+            for (OnEventCreatedListener listener : onEventCreatedListeners)
+                listener.onEventCreated(event);
+
+        Marker created = mMap.addMarker(new MarkerOptions().position(pos).title(description)); //automatically cuts title if too long
+        created.setTag(event);
+        moveMap(pos);
+    }
+
+    /**
+     * Called when user long presses an InfoWindow of Marker
+     *
+     * @param marker The marker whose InfoWindow was pressed
+     */
+    @Override
+    public void onInfoWindowLongClick(Marker marker) {
+        currentFocusMarker = marker;
+        callRemoveEventDialog(marker);
+    }
+
+    /**
+     * Opens a dialog that asks the user if the event has to be deleted
+     * If the user accepts, the dialog will call RemoveMark method of this instance
+     *
+     * @param marker The marker that has been selected for confirming deletion
+     */
+    protected void callRemoveEventDialog(Marker marker) {
+        if (removeEventDialog != null) {
+            removeEventDialog.setMarker(marker);
+            removeEventDialog.show(supportFragmentManager, REMOVE_EVENT_DIALOG_TAG);
+        }
+    }
+
+    /**
+     * Called by the RemoveEventDialog or the bottom sheet if the user confirms to delete the event
+     * Removes the Event mark from the map
+     * Makes forget the previously set currentFocusMarker
+     * Calls all the onEventRemovedListeners

```

Comment:
```suggestion
     * Called by the {@link #RemoveEventDialog} method or the bottom sheet if the user confirms to delete the event
     * Removes the Event mark from the map
     * Makes forget the previously set currentFocusMarker
     * Calls all the onEventRemovedListeners
```

You can "spam" {@link ...} everywhere or wherever you think it's necessary, but you are not obligated.
I want to say to you that I prefer to spam it everywhere because is always a _fast and simply reference_.

Reply:
Makes sense👌